File: Leetcode/CPP/0347.cpp
#include<iostream>
#include<map>
#include<vector>

using namespace std;

vector<int> topKFrequent(vector<int>& nums, int k) {
        vector<int> ans;
        
        return ans;
}

int main(){
    vector<int> nums={1,1,1,2,2,3};
    int k=2;
    vector<int> ans= topKFrequent(nums,k);
    for (int i=0;i<ans.size();i++){
        cout<<ans[i]<<", ";
    }
    cout<<endl;
    return 0;
}

File: Leetcode/GO/0347.go
package main

// func main() {
// 	nums := []int{1, 1, 1, 2, 2, 3}
// 	k := 2
// }

func topKFrequent(nums []int, k int) []int {

	countMap := map[int]int{}
	for _, num := range nums {
		if count, ok := countMap[num]; ok {
			countMap[num] = count + 1
		} else {
			countMap[num] = 1
		}
	}

	countSlice := make([][]int, len(nums)+1)
	for num, count := range countMap {
		countSlice[count] = append(countSlice[count], num)
	}

	res := []int{}
	for i := len(countSlice) - 1; i > 0; i-- {
		res = append(res, countSlice[i]...)
		if len(res) == k {
			return res
		}
	}
	return res
}
